### DISCUSSION TITLE
Teamtalk server crashing

### DISCUSSION THREAD

**Discussion by DalisonCPU** [BearWare/TeamTalk5#2071] [Q&A]
### Description

For some reason, when creating a channel, the Teamtalk server crashes.
Settings this happens:
Server Version:
5.13.1.5115

Version of the client I'm using to create the channel:
TeamTalk 5 Classic v. 5.12
P.S: I believe this has to do with characters related to emogis, probably due to the amount of bits each one uses.

### Application

- [ ] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [X] TeamTalkClassic
- [X] TeamTalkServer

### Platform

- [X] Windows
- [ ] macOS
- [ ] Android
- [ ] iOS
- [X] Linux

### Expected behavior

The server should work normally

### Actual behavior

When I create a channel, the server goes down.
P.S: The channel is created even if the server goes down, when it is active I automatically join the channel I created.

### Steps to reproduce problem

1. With a user account that has permission to create permanent channels, open the menu to create a channel
2. Enter the channel name, a password and check permanent channel.
Change from mono to stereo.
Change the transmission interval to 20.
Uncheck the box to join the channel (I don't know the right name in English, but it's the last box to check shown in the options.
Click create channel and I believe the server will crash.

**Comment by amirmahdifard** [comment]
what do you enter as channel name and password, since you said it has to do with emoji: it can help out to reproduce more bettar.

**Comment by DalisonCPU** [comment]
I don't know how to explain it properly. I can put any name that the server drops, but I tested it on some servers and it doesn't happen.
In my case, there is a channel called 🗣🔐
I create any channel with a name inside it and the server goes down. It could be friends, VIP friends etc... Everything takes the server.
If you send me an email privately I can send you the .xml file for you to analyze.

**Comment by bear101** [comment]
I cannot reproduce the issue.

**Comment by DalisonCPU** [comment]
How can I send the .xml for you to test on a server?

**Comment by DalisonCPU** [comment]
@bear101

**Reply by bear101** [reply]
does the xml-file you have crash the server? I've already tried to reproduce the channel-name-error but I don't see any crashes.

**Comment by DalisonCPU** [comment]
@bear101 I'm thinking so

**Reply by bear101** [reply]
Think so? Does it crash or not upon startup?

**Reply by DalisonCPU** [reply]
No, it doesn't crash on startup. It only crashes when you update a channel, or update the server. I think it must have something to do with character incompatibility.
I honestly don't know what it could be. I thought it could also be an old configuration version, since this configuration file is quite old (more than two years).
My suggestion is that I send you the configuration file privately and that a debug version be made, so you can test more precisely what may be happening.
I remembered something very important. With a new configuration file this no longer happens, but if I update a channel with the classic version of the client or the server properties, the client closes completely.

**Reply by bear101** [reply]
Please create a configuration file that crashes as soon as the server is started. If there's no reliable way to reproduce the problem then I cannot fix it.

**Comment by amirmahdifard** [comment]
@DalisonCPU if you can, send me your xml file to my email amirmahdifard0@gmail.com and i will read it line by line in notepad and will compare it with my own new config, and will see what makes it to crash: i will say the result here. Hopefully i willl find out the crash: i'm glad to help!

**Comment by DalisonCPU** [comment]
Hello @bear101 @amirmahdifard.
I have new information. Teamtalk is experiencing a security error, but it is not related to the configuration file.
The server always crashes when someone connects to the server, so it's clearly a security flaw. I would like to request a debug version or something that allows you to activate Teamtalk debugging, which notes all commands sent to the server before they are executed to a file, so that it is possible to identify what failure is occurring.
Anyway, I'll send the log so you can see. Teamtalk was crashing when user #81 connected.
[tt5srv.log](https://github.com/BearWare/TeamTalk5/files/13260635/tt5srv.log)

**Comment by DalisonCPU** [comment]
@bear101 @amirmahdifard Please, I would like an answer. While you don't fix it, the server is being taken offline all the time because of this error.

**Comment by bear101** [comment]
Have you tried to update to v5.14. It fixes a crash issue in the server.

**Reply by DalisonCPU** [reply]
Yes. The version of server is 5.15.0.5132

**Reply by Ramon-Salazar** [reply]
There is another crash issue that isn't yet fixed, one of my friend's server got crashed twice when some suspicious users loged in despite that he's running v5.14.0.5128 probably by getting an account which has auto-operator rights, so I guess there is another security Vulnerability in some expressions probably by using custom client either kick or ban function.

The bug is still exist in the beta version too, v5.15.0.1532, since I remembered the same suspicious user with the noname loged in and crashed another unofficial server.

**Comment by DalisonCPU** [comment]
I thought it might be with some option sent when creating a channel
